### PR TITLE
Codemod v3 slots improvements

### PR DESCRIPTION
**Pull request by kirillplatonov**
This PR includes improvements for #1669:

- Adds rake task to detect legacy slots `bin/rails view_component:detect_legacy_slots`
- Adds rake to migrate legacy slots automatically `bin/rails view_component:migrate_legacy_slots`
- Adds component previews to codemods view paths
- Autodetection for Rails view paths
- Improves uncertain matches detection
- Updated changelog

### CODE REVIEW

File: lib/view_component/codemods/v3_slot_setters.rb
Comment (Spone) [inline lib/view_component/codemods/v3_slot_setters.rb:10]: If we join to Rails' root, I guess we don't need the `..\`

```suggestion
#       view_path: Rails.root.join("app/views"),
```

File: docs/CHANGELOG.md
Comment (Spone) [inline docs/CHANGELOG.md:21]:
```suggestion
* Add codemod to detect and migrate deprecated Slots setters to new `with_*` prefix introduced in v3.0.0.
```

File: lib/view_component/codemods/v3_slot_setters.rb
Comment (Spone) [inline lib/view_component/codemods/v3_slot_setters.rb:24]: If I remember correctly, I used `Zeitwerk` directly so the codemod works in gems that do no require Rails (this needs to be verified).

File: lib/view_component/codemods/v3_slot_setters.rb
Comment (Spone) [inline lib/view_component/codemods/v3_slot_setters.rb:203]: Does this break when running the codemod in a gem that does not require Rails?

File: docs/CHANGELOG.md
Comment (Spone) [reply docs/CHANGELOG.md:25]: Also, maybe we can pass custom paths to the task, similar to rubocop:

```
rubocop          # default
rubocop app lib  # run on specific paths
```

Comment (kirillplatonov) [reply docs/CHANGELOG.md:25]: In rake tasks we can use `--fix` option only with the options separator:
```bash
rails view_component:detect_legacy_slots --- --fix
```
Not sure if it's cleaner than a separate rake task for migrations.

Comment (kirillplatonov) [reply docs/CHANGELOG.md:25]: > Also, maybe we can pass custom paths to the task, similar to rubocop:
Added 👍

File: lib/view_component/codemods/v3_slot_setters.rb
Comment (kirillplatonov) [reply lib/view_component/codemods/v3_slot_setters.rb:25]: This will cause issues with apps that don't use `Zeitwerk`. Eg some versions of Rails in the `view_component` test suite will fail. The library is Rails-only so I don't think it should be a problem.

File: lib/view_component/codemods/v3_slot_setters.rb
Comment (kirillplatonov) [reply lib/view_component/codemods/v3_slot_setters.rb:204]: In gems app rake tasks require dummy app and at least development rails dependency:
```
# Rakefile
APP_RAKEFILE = File.expand_path("test/dummy/Rakefile", __dir__)
load "rails/tasks/engine.rake"
```
And can be executed with the prefix:
```
bin/rails app:view_component:detect_legacy_slots
```
Considering that, I don't think Rails view paths call will be a problem.

**Comment by kirillplatonov** [comment]
@Spone can we merge it at the current state? I included this class in Polaris ViewComponents v1 release and tested upgrade on a couple production apps. It works great. 

Even if the current implementation is not perfect it will be beneficial to have it shipped with ViewComponent now when people are doing upgrades to v3. In a couple of months this will be less useful I suppose.

**Comment by Spone** [comment]
I agree. What do you think @joelhawksley?

**Review by joelhawksley — COMMENTED** [review]
I like the idea of getting this into `main` even if it isn't perfect ❤️

File: docs/CHANGELOG.md
Comment (kirillplatonov) [reply docs/CHANGELOG.md:25]: Added a note to changelog. Not sure where to put info about this codemod in docs. It seems like there's no upgrade section at the moment.

File: Gemfile
Comment (kirillplatonov) [reply Gemfile]: Removed it

**Comment by kirillplatonov** [comment]
@joelhawksley please note that this PR is a contribution to #1669. We'll need to make a final review there once merged.

**Review by Spone — APPROVED** [review]

**Comment by Spone** [comment]
Let's wrap this up in #1669!

**Pull request merged**
